file/org.wso2.carbon.transport.remote-file-system/src/main/java/org/wso2/carbon/transport/remotefilesystem/server/RemoteFileSystemConsumer.java

```diff
@@ -101,42 +99,53 @@ public RemoteFileSystemConsumer(String id, Map<String, String> fileProperties,
             }
             FileType fileType = getFileType(listeningDir);
             if (fileType != FileType.FOLDER) {
-                String errorMsg = "File system server connector is used to " +
+                String errorMsg = "[" + serviceName + "] File system server connector is used to " +
                         "listen to a folder. But the given path does not refer to a folder.";
-                remoteFileSystemListener.onError(new RemoteFileSystemConnectorException(errorMsg));
-                throw new ServerConnectorException(errorMsg);
+                final RemoteFileSystemConnectorException exception
+                        = new RemoteFileSystemConnectorException(errorMsg);
+                remoteFileSystemListener.onError(exception);
+                throw exception;
             }
             //Initialize the thread executor based on properties
-            threadPool = new ThreadPoolFactory(threadPoolSize, parallelProcess);
-        } catch (FileSystemException | RemoteFileSystemConnectorException e) {
+            threadPool = new ThreadPoolFactory(threadPoolSize);
```

Comment:
out of curiosity. in a thread dump is there a way to identify these threads? ( named threads ?)

Reply:
This was fixed by adding a custom Thread Factory implementation.